index.js

```diff
@@ -49,17 +49,22 @@ function restrictTabBehavior(event: KeyboardEvent): void {
   if (elements.length === 0) return
 
   const movement = event.shiftKey ? -1 : 1
-  const currentFocus = elements.filter(el => el.matches(':focus'))[0]
-  let targetIndex = 0
+  const currentFocus = dialog.contains(document.activeElement) ? document.activeElement : null
+  let targetIndex = movement === -1 ? -1 : 0
 
   if (currentFocus) {
     const currentIndex = elements.indexOf(currentFocus)
     if (currentIndex !== -1) {
-      const newIndex = currentIndex + movement
-      if (newIndex >= 0) targetIndex = newIndex % elements.length
```

Comment:
When `newIndex` is `-1` (`(currentIndex)0 + (movement)-1`), `targetIndex` stays `0`.